### PR TITLE
Remove dead code and overhaul the framework

### PR DESCRIPTION
**Pull request by phoe**
## What

This PR shaves about 660 lines off the size of the repository and refactors the testing and printing frameworks.

## How

* We highly customize the attached `lisp-unit` test framework and gut out everything that is not required to work with the koans. A side effect is that verifying the koans should now be slightly faster.
* We refactor the file `contemplate.lsp` into two files: `lisp-koans.lsp` that defines the package and functions related to koans and `contemplate.lsp` which actually loads the code and executes the main function. 

## Why

* The much smaller code of the testing and printing framework is more maintainable due to dead code removal.
* The above will allow lisp-koans to be loadable via ASDF in the future, and it will be possible to evaluate koans' completion via the Lisp REPL instead of using shell scripts.
* This PR will also allow to include lisp-koans in Quicklisp in the future. The internal package `lisp-unit` was renamed and therefore no longer collides with the Quicklisp-providable `lisp-unit` system.

## To be done

* [ ] code review
* [ ] test completing the koans with the refactored koan framework

### CODE REVIEW

File: lisp-koans.lsp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+;;;
+;;; Unless required by applicable law or agreed to in writing, software
+;;; distributed under the License is distributed on an "AS IS" BASIS,
+;;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;;; See the License for the specific language governing permissions and
+;;; limitations under the License.
+
+(defpackage #:lisp-koans
+  (:use #:common-lisp
+        #:com.google.lisp-koans.test
```
Comment (Slids) [inline lisp-koans.lsp]: again can't use com.google.

File: lisp-unit.lsp [hunk 2]
```diff
@@ -1,5 +1,3 @@
-;;; -*- Mode: Lisp; Syntax: ANSI-Common-Lisp -*-
```
Comment (Slids) [inline lisp-unit.lsp]: You did this someplaces but not all it seems...

File: lisp-koans.lsp [hunk 3]
```diff
@@ -41,7 +41,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 (defun package-name-from-group-name (group-name)
-  (format nil "COM.GOOGLE.LISP-KOANS.KOANS.~:@(~A~)" group-name))
+  (format nil "COM.GOOGLE.LISP-KOANS.KOANS.~A" group-name))
```
Comment (Slids) [inline lisp-koans.lsp:31]: Have to remove com.google here too... Could you do that?

File: contemplate.lsp [hunk 4]
```diff
@@ -27,9 +27,9 @@
 (load "lisp-unit.lsp")
 
 (defpackage :lisp-koans
-  (:use :common-lisp)
-  (:use :lisp-unit)
-  #+sbcl (:use :sb-ext))
+  (:use #:common-lisp
+        #:com.google.lisp-koans.test
```
[the remark this answers is not in the document]
Comment (phoe) [reply contemplate.lsp]: OK. Which package name should we instead use?

File: lisp-koans.lsp (hunk 1 above)
Comment (phoe) [reply lisp-koans.lsp]: Ditto.

File: lisp-koans.lsp (hunk 3 above)
Comment (phoe) [reply lisp-koans.lsp:31]: Ditto.

File: lisp-unit.lsp (hunk 2 above)
Comment (phoe) [reply lisp-unit.lsp]: What do you mean? I cannot see any other modelines in the repository:

```bash
~/Projects/Lisp/lisp-koans $ git grep ANSI-Common-Lisp
~/Projects/Lisp/lisp-koans $
```

**Comment by phoe** [comment]
This was not tested yet; I'm still fixing up the koans and I'll then want to test everything together.

**Comment by phoe** [comment]
Closing this for now - I will submit a more complete and tested PR soon.

**Pull request closed without merging**